src/main/java/cofh/thermaldynamics/duct/nutypeducts/TileGrid.java

```diff
@@ -1243,7 +1243,7 @@ public boolean canConnectEnergy(EnumFacing from) {
 	}
 
 	@Override
-	public boolean hasCapability(@Nonnull Capability<?> capability, @Nonnull EnumFacing facing) {
+	public boolean hasCapability(@Nonnull Capability<?> capability, @Nullable EnumFacing facing) {
```

Comment:
context: Capabilities are not only used for tile entities but also for entities, itemstacks, worlds and so on. Since all of these capability holders dont have a use for a facing value, the value is nullable.